pulp_python/app/utils.py

```diff
@@ -60,7 +66,147 @@ def parse_metadata(project, version, distribution):
     package['version'] = version
     package['url'] = distribution.get('url') or ""
     package['sha256_digest'] = distribution.get('digests', {}).get('sha256') or ""
+    package['python_version'] = distribution.get('python_version') or ""
 
     package.update(parse_project_metadata(project))
 
     return package
+
+
+def python_content_to_json(base_path, content_query, version=None):
+    """
+    Converts a QuerySet of PythonPackageContent into the PyPi JSON format
+    * Find the PEP *
+    JSON metadata has:
+        info: Dict
+        last_serial: int
+        releases: Dict
+        urls: Dict
+    """
+    full_metadata = {"last_serial": 0}  # For now the serial field isn't supported by Pulp
+    latest_content = latest_content_version(content_query, version)
+    full_metadata.update({"info": python_content_to_info(latest_content[0])})
+    full_metadata.update({"releases": python_content_to_releases(content_query, base_path)})
+    full_metadata.update({"urls": python_content_to_urls(latest_content, base_path)})
+    return full_metadata
+
+
+def latest_content_version(content_query, version):
+    """
+    Walks through the content QuerySet and finds the instances that is the latest version.
+    If 'version' is specified, the function instead tries to find content instances
+    with that version and raises an exception if not found
+    """
+    latest_version = version
+    latest_content = []
+    for content in content_query:
+        if version and parse(version) == parse(content.version):
+            latest_content.append(content)
+        elif not latest_version or parse(content.version) > parse(latest_version):
+            latest_content = [content]
+            latest_version = content.version
+        elif parse(content.version) == parse(latest_version):
+            latest_content.append(content)
+
+    if version and not latest_content:
+        raise Exception()
+    return latest_content
+
+
+def python_content_to_info(latest_content):
+    """
+    Takes in a PythonPackageContent instance and returns a dictionary of the Info fields
+    """
+    # Would have been nice to use a serializer,
+    # but I couldn't figure out how to pass in the request context
+    info = dict(latest_content.__dict__)
```

Comment:
TBH, I think it might be easier (and cleaner) to just manually create your dict with the attributes you want to add, rather than grabbing the internal ```__dict__``` and removing stuff.